benchmark-automation/requirements.md

```diff
@@ -0,0 +1,125 @@
+# Faster CPython benchmarking automation infrastructure
+
+This is a requirements gathering brainstorm exercise for the benchmarking infrastructure for the Faster CPython project.
+
+## Goals
+
+To make anyone contributing to make CPython faster more efficient and effective by automating away some of the tedious or error-prone tasks related to benchmarking.
+
+Visualization of the results should be designed to understand the nuance of specific performance improvements.
+
+Growing and improving the benchmark suite is going to be important over time.
+There should be an easy way to regenerate baselines as the suite is evolved so that existing results are not invalidated.
+
+The wider Python community should be able to easily track the progress of CPython performance.
+
+## Guardrail Requirements
+
+Ensure that this infrastructure is a community resource: the results are visible by everyone, and usable and maintainable by anyone with a demonstrated interest, regardless of their employment status.
+
+All parts of the system should be replicatable locally at the command line, and not require any complicated setup.
+
+## Out of scope
+
+We aren't aiming to support an "implementation shootout", e.g. between PyPy, Pyston and CPython.
+The pyperformance benchmark suite will continue to be designed for this, but the rest of the tooling described here may not necessarily support it.
+
+## Block diagram
+
+The following is a block diagram of the proposed system.
+
+While this is not intended as a description of current functionality, existing tools that provide some of this functionality are noted.
+Importantly, there isn't currently a connection between the benchmark runner and `speed.python.org` -- these are currently distinct efforts.
+
+![Block diagram](/benchmark-automation/Benchmark Infrastructure Block Diagram.png)
+
+## Definitions
+
+### Results
+
+"Results", all of which come from running `pyperformance` in the same configuration, come in the following flavors:
+
+- **Nightly**: Run once a day on the current `main` branch of CPython.
+- **Tag**: Run on tagged releases of Python. The most recent patch release for all supported releases, and all alphas, betas and rcs for the current development release.
```

Comment:
Maybe clarify? Between the May beta freeze and the October release each year, we have two different branches (currently, `3.11` and `main`) that could be called a "current development release".

Since not too much performance development happens after the beta freeze, perhaps this can be simplified to "alphas from the current `main` cycle" or similar?

```suggestion
- **Tag**: Run on tagged releases of Python. The most recent tag for all feature branches and all alphas from the current `main` cycle.
```

Currently, this would include:
- `v3.7.13`
- `v3.8.13`
- `v3.9.13`
- `v3.10.4`
- `v3.11.0b1`

(3.12 isn't represented yet, since no alphas have been tagged.)